accounts-db/src/accounts_index/in_mem_accounts_index.rs

```diff
@@ -919,11 +941,23 @@ impl<T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> InMemAccountsIndex<T,
         let (possible_evictions, m) = {
             let map = self.map_internal.read().unwrap();
             let m = Measure::start("flush_scan"); // we don't care about lock time in this metric - bg threads can wait
+
+            // Calculate max evictions for threshold-based flushing
+            let max_evictions = if !startup {
+                let entries_in_bin = map.len();
+                // Low water mark: flush down to 50% of limit
+                self.storage
+                    .max_evictions_for_threshold(entries_in_bin, 0.50)
+            } else {
+                None
+            };
```

Comment:
Shouldn't matter for startup or not. At startup, nothing is in the in-mem hashmaps.